cassovary-core/src/main/scala/com/twitter/cassovary/graph/ArrayBasedDirectedGraph.scala

```diff
@@ -291,10 +323,30 @@ object ArrayBasedDirectedGraph {
         }
       }
 
+      def finishInEdgesFilling(): Future[Unit] = {
+        log.debug("finishing filling")
+        statsReceiver.time("finishing_filling_in_edges") {
+          val futures = nodesOutEdges.map {
+            nodes => futurePool {
+              nodes.foreach {
+                node =>
+                  table(node.id) match {
+                    case fillingInEdgesNode: FillingInEdgesBiDirectionalNode =>
+                      table(node.id) = fillingInEdgesNode.finishingFilling(neighborsSortingStrategy != LeaveUnsorted)
+                    case _ => // no need to finish filling
+                  }
+              }
+            }
+          }
+          Future.join(futures)
+        }
+      }
+
       for {
         inEdgesSizes <- findInEdgesSizes(nodesOutEdges, nodeIdSet, numNodes)
         _ <- instantiateInEdges(inEdgesSizes)
         _ <- populateInEdges(inEdgesSizes)
+        _ <- when(neighborsSortingStrategy != LeaveUnsorted) (finishInEdgesFilling())
```

Comment:
one effect of this condition is of course that under this sorting strategy, nodes are of type FillingInEdgesBiDirectionalNode are left as it is (which is okay with me as table[] is of type Array[Node] but you had pointed out earlier that this was not ideal)